cluster-autoscaler/core/scaleup/orchestrator/orchestrator.go

```diff
@@ -391,20 +391,30 @@ func (o *ScaleUpOrchestrator) ScaleUpToNodeGroupMinSize(
 		}
 
 		if skipReason := o.IsNodeGroupResourceExceeded(resourcesLeft, ng, nodeInfo); skipReason != nil {
-			klog.Warning("ScaleUpToNodeGroupMinSize: node group resource excceded: %v", skipReason)
+			klog.Warningf("ScaleUpToNodeGroupMinSize: node group resource excceded: %v", *skipReason)
 			continue
 		}
 
 		newNodeCount := ng.MinSize() - targetSize
 		newNodeCount, err = o.resourceManager.ApplyLimits(o.autoscalingContext, newNodeCount, resourcesLeft, nodeInfo, ng)
 		if err != nil {
-			klog.Warning("ScaleUpToNodeGroupMinSize: failed to apply resource limits: %v", err)
+			klog.Warningf("ScaleUpToNodeGroupMinSize: failed to apply resource limits: %v", err)
 			continue
 		}
 
 		newNodeCount, err = o.GetCappedNewNodeCount(newNodeCount, targetSize)
```

Comment:
One thing that I just noticed (and is a separate bug):
`IsNodeGroupResourceExceeded` is expecting the **total** size of the **cluster** as the second parameter. 
As it checks the total cluster size.
See the line in the standard scaleup: `GetCappedNewNodeCount(bestOption.NodeCount, len(nodes)+len(upcomingNodes))`

Once this is addressed it should also use an approach similar to `AllocateResources`

Reply:
I'm assuming you were talking about `GetCappedNewNodeCount`. Yes, you are right, I just pushed the fix and added test cases to cover this. Thanks for pointing this out.